xbmc/video/guilib/VideoStreamSelectHelper.cpp

```diff
@@ -0,0 +1,480 @@
+constexpr int STREAM_ID_NONE = -1; // Stream id referred to item for "none" stream
+
+// \brief Make a FileItem entry with "Disable" label, allow to disable a stream
+const std::shared_ptr<CFileItem> MakeFileItemDisable(bool isSelected)
+{
+  const auto fileItem{std::make_shared<CFileItem>(g_localizeStrings.Get(24021))};
+  fileItem->Select(isSelected);
+  fileItem->SetProperty("stream.id", STREAM_ID_DISABLE);
+  return fileItem;
+}
+
+// \brief Make a FileItem entry with "None" label, signal an empty list
+const std::shared_ptr<CFileItem> MakeFileItemNone()
+{
+  const auto fileItem{std::make_shared<CFileItem>(g_localizeStrings.Get(231))};
+  fileItem->SetProperty("stream.id", STREAM_ID_NONE);
+  return fileItem;
+}
+
+std::shared_ptr<const CFileItem> OpenSelectDialog(CGUIDialogSelect& dialog,
+                                                  int headingId,
+                                                  const CFileItemList& itemsToDisplay)
+{
+  dialog.Reset();
+  dialog.SetHeading(headingId);
+  dialog.SetUseDetails(true);
+  dialog.SetMultiSelection(false);
+  dialog.SetItems(itemsToDisplay);
+
+  dialog.Open();
+
+  if (dialog.IsConfirmed())
+    return dialog.GetSelectedFileItem();
+
+  return {};
+}
+
+std::string CovertFpsToString(float value)
+{
+  if (value == 0.0f)
+    return "";
+
+  std::string str{StringUtils::Format("{:.3f}", value)};
+  // Keep numbers after the comma only if they are not 0
+  const size_t zeroPos = str.find_last_not_of("0");
+  if (zeroPos != std::string::npos)
+    str.erase(zeroPos + 1);
+
+  if (str.back() == '.')
+    str.pop_back();
+
+  return str;
+}
+
+struct VideoStreamInfoExt : VideoStreamInfo
+{
+  VideoStreamInfoExt(int id, const VideoStreamInfo& info) : VideoStreamInfo(info)
+  {
+    streamId = id;
+    isDefault = info.flags & StreamFlags::FLAG_DEFAULT;
+    isForced = info.flags & StreamFlags::FLAG_FORCED;
+    isHearingImpaired = info.flags & StreamFlags::FLAG_HEARING_IMPAIRED;
+    isVisualImpaired = info.flags & StreamFlags::FLAG_VISUAL_IMPAIRED;
+  }
+
+  int streamId;
+  std::string languageDesc;
+  bool isDefault;
+  bool isForced;
+  bool isHearingImpaired;
+  bool isVisualImpaired;
+};
+
+struct AudioStreamInfoExt : AudioStreamInfo
+{
+  AudioStreamInfoExt(int id, const AudioStreamInfo& info) : AudioStreamInfo(info)
+  {
+    streamId = id;
+
+    if (!g_LangCodeExpander.Lookup(info.language, languageDesc))
+      languageDesc = g_localizeStrings.Get(13205); // Unknown
+
+    isDefault = info.flags & StreamFlags::FLAG_DEFAULT;
+    isForced = info.flags & StreamFlags::FLAG_FORCED;
+    isHearingImpaired = info.flags & StreamFlags::FLAG_HEARING_IMPAIRED;
+    isVisualImpaired = info.flags & StreamFlags::FLAG_VISUAL_IMPAIRED;
+    isOriginal = info.flags & StreamFlags::FLAG_ORIGINAL;
+  }
+
+  int streamId;
+  std::string languageDesc;
+  bool isDefault;
+  bool isForced;
+  bool isHearingImpaired;
+  bool isVisualImpaired;
+  bool isOriginal;
```

Comment:
I would init the bools and the int here, for safety.

Reply:
done